apps/browser/src/platform/offscreen-document/offscreen-document.service.ts

```diff
@@ -0,0 +1,46 @@
+export class DefaultOffscreenDocumentService implements DefaultOffscreenDocumentService {
+  private workerCount = 0;
+
+  constructor() {}
+
+  async withDocument<T>(
+    reasons: chrome.offscreen.Reason[],
+    justification: string,
+    callback: () => Promise<T> | T,
+  ): Promise<T> {
+    this.workerCount++;
+    try {
+      if (!(await this.documentExists())) {
+        await this.create(reasons, justification);
+      }
+
+      const maybePromise = callback();
+
+      if (maybePromise instanceof Promise) {
+        return await maybePromise;
+      }
+      return maybePromise;
```

Comment:
```suggestion
      const promiseValue = callback();
      
      return await Promise.resolve(promiseValue);
```

I might be wrong about this, but would the above suggestion be equivalent?

Reply:
Can't you just await the value directly?

```ts
return await callback();
```

I think you can await anything, even regular values

![image](https://github.com/bitwarden/clients/assets/2285588/f9c4481c-24f5-4636-9425-9ed508626e91)

Reply:
And so it's reduced to the minimum, thank you 😄

Reply:
This chain did get me to google why I do my pattern again, and it's [this](https://stackoverflow.com/a/55263084).

If it's not a promise, it seems better to me to not await it to maintain execution order as much as I can. Here it probably doesn't matter because there are other async methods being awaited, but in general I don't think `await` or `Promise.resolve` is a good replacement for type checking and awaiting when necessary.